### PR TITLE
Add function to compute Quaternion given two SVector{3}s

### PR DESCRIPTION
**Pull request by tkoolen**
Adds a constructor for `AngleAxis` given two vectors, `from` and `to`, such that applying the constructed rotation to `from` results in `to` (up to scaling). I needed this and thought it might be useful for others as well.

### CODE REVIEW

**Comment by c42f** [comment]
Thanks for this, I think it's very useful functionality to have in `Rotations`.

Perhaps it would it be better arranged as a function rather than a constructor? I guess it depends on the user's point of view as to whether they really want an `AngleAxis` from this or any kind of rotation will do. `rotate_between(a,b)` might work as a name which is likely to be somewhat readable and intuitive at the call site? Is there a better verb for generating a rotation which links two geometric objects?

As a side note, the rotation between two vectors is very non-unique, with the two vectors pinning down only two out of three degrees of freedom. I think this is kind of covered by your note in the documentation but not exactly?  Not that I disagree with the implementation, it looks clearly like the right thing.

**Comment by andyferris** [comment]
This looks incredibly useful - thanks @tkoolen.

**Comment by tkoolen** [comment]
> function rather than a constructor

That sounds good to me. I was doubting between those two options myself. Maybe `rotation_between(from, to)` is slightly clearer than `rotate_between`? For comparison:

* Ogre uses [`OgreVector3::getRotationTo`]( https://bitbucket.org/sinbad/ogre/src/9db75e3ba05c/OgreMain/include/OgreVector3.h?fileviewer=file-view-default#OgreVector3.h-651)
* Unity uses [`Quaternion::FromToRotation`](https://docs.unity3d.com/ScriptReference/Quaternion.FromToRotation.html)
* Eigen uses [`Quaternion::FromTwoVectors`](http://eigen.tuxfamily.org/dox/classEigen_1_1Quaternion.html)

Note by the way that all of these implementations return quaternions; I do think now that computing a quaternion might be faster, especially since quaternions are typically more useful for subsequent operations. I'll probably port [Eigen's implementation](https://bitbucket.org/eigen/eigen/src/60f8b0f18340337e036b566bc583c77deb52f965/Eigen/src/Geometry/Quaternion.h?at=default&fileviewer=file-view-default#Quaternion.h-578) and compare.

> As a side note, the rotation between two vectors is very non-unique, with the two vectors pinning down only two out of three degrees of freedom.

The third degree of freedom is determined by the requirement that the rotation have minimal angle (i.e. the shortest-path rotation). I did mention that in the docstring, but let me know if you think more documentation is necessary.

File: src/angleaxis_types.jl [hunk 1]
```diff
@@ -29,6 +29,38 @@ end
...
+    else
+        axis = cross(from, to)
+        if dot(axis, axis) < 10 * eps(T)^2
+            # 'from' and 'to' point in opposite directions.
+            # find axis that is perpendicular to 'from' and 'to'
+            maxnormvec = normfrom > normto ? from : to
+            axis = perpendicular_vector(maxnormvec)
+        end
+        # note: relies on normalization of axis in AngleAxis constructor:
+        AngleAxis(ϕ, axis[1], axis[2], axis[3])
```
Comment (andyferris) [inline src/angleaxis_types.jl]: `@inbounds return AngleAxis(...)`

File: src/util.jl [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+largest absolute value, flipping the sign of the second largest, and setting the
+remaining element to zero.
+"""
+function perpendicular_vector(vec::SVector{3})
+    T = eltype(vec)
+
+    # find indices of the two elements of vec with the largest absolute values:
+    absvec = abs(vec)
+    ind1 = indmax(absvec) # index of largest element
+    absvec2 = @SVector [ifelse(i == ind1, typemin(T), absvec[i]) for i = 1 : 3] # set largest element to typemin(T)
```
Comment (andyferris) [inline src/util.jl]: Might need `@inbounds` here too

File: src/util.jl [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    ind2 = indmax(absvec2) # index of second-largest element
+
+    # perp[ind1] = -vec[ind2], perp[ind2] = vec[ind1], set remaining element to zero:
+    perpind1 = -vec[ind2]
+    perpind2 = vec[ind1]
+    perp = @SVector [ifelse(i == ind1, perpind1, ifelse(i == ind2, perpind2, zero(T))) for i = 1 : 3]
```
Comment (andyferris) [inline src/util.jl]: And here.

Comment (andyferris) [inline src/util.jl]: No, sorry, that's wrong! It's the two lines above.

**Comment by tkoolen** [comment]
OK, switched from `AxisAngle` constructor to `rotation_between` function that returns a quaternion and addressed comments.

Things actually work out a lot nicer when you use quaternions. I ended up basing my implementation on [this forum post](https://www.gamedev.net/topic/429507-finding-the-quaternion-betwee-two-vectors/#entry3856228), after [rederiving the math myself](https://github.com/FugroRoames/Rotations.jl/files/709982/geodesic-quaternion.pdf). The reason I went with (a robustified version of) this implementation instead of the ones I posted earlier is that it exploits the normalization that already happens in the `Quat` constructor. Let me know if you have any questions.

As for the name `rotation_between`, it's alright, but I still don't like it that much. Let me know if you have a better name. Maybe `shortest_path_quaternion`?

**Comment by andyferris** [comment]
I can't think of anything better right now. It returns a `Rotation` so `rotation_between` seems to make sense, its just if we can find a prettier name that might be better.

We can change and deprecate this later if we think of a better name.

**Comment by c42f** [comment]
Agreed, thanks @tkoolen!

**Comment by tkoolen** [comment]
Thanks, guys!

**Pull request merged**